packages/server/postgres/migrations/1635354572171_addIntegrationProviderAndToken.ts

```diff
@@ -0,0 +1,91 @@
+import {Client} from 'pg'
+import getPgConfig from '../getPgConfig'
+
+export async function up() {
+  const client = new Client(getPgConfig())
+  await client.connect()
+  await client.query(`
+  DO $$
+  BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'IntegrationProviderTypesEnum') THEN
+      CREATE TYPE "IntegrationProviderTypesEnum" AS ENUM (
+        'gitlab',
+        'mattermost'
+      );
+    END IF;
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'IntegrationProviderTokenTypeEnum') THEN
+      CREATE TYPE "IntegrationProviderTokenTypeEnum" AS ENUM (
+        'pat',
+        'oauth2',
+        'webhook'
+      );
+    END IF;
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'IntegrationProviderScopesEnum') THEN
+      CREATE TYPE "IntegrationProviderScopesEnum" AS ENUM (
+        'global',
+        'org',
+        'team'
+      );
+    END IF;
+    CREATE TABLE IF NOT EXISTS "IntegrationProvider" (
+      "id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+      "createdAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+      "updatedAt" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP NOT NULL,
+      "type" "IntegrationProviderTypesEnum" NOT NULL,
+      "tokenType" "IntegrationProviderTokenTypeEnum" NOT NULL,
+      "scope" "IntegrationProviderScopesEnum" NOT NULL,
+      "scopeGlobal" BOOLEAN GENERATED ALWAYS AS (
+        CASE
+          WHEN "scope" = 'global' THEN TRUE
+          ELSE NULL
+        END
+      ) STORED,
+      "orgId" VARCHAR(100),
+      "teamId" VARCHAR(100),
+      "isActive" BOOLEAN DEFAULT TRUE NOT NULL,
+      "name" VARCHAR(250) NOT NULL,
+      "providerMetadata" JSONB NOT NULL DEFAULT '{}',
+      UNIQUE("scopeGlobal", "type"),
+      CONSTRAINT global_provider_must_be_oauth2 CHECK (
+        "scopeGlobal" IS NULL OR ("scopeGlobal" = TRUE AND "tokenType" = 'oauth2')
+      )
+    );
+    CREATE INDEX IF NOT EXISTS "idx_IntegrationProvider_typeAndScope"
```

Comment:
i couldn't find a use for this, given it is coming from a user we'd have their token, which means we'd have the providerId, so we shouldn't need this